st2common/st2common/services/executions.py

```diff
@@ -42,6 +44,22 @@
 
 LOG = logging.getLogger(__name__)
 
+SKIPPED = ['id', 'context', 'callback', 'action']
+
+
+def _decompose_liveaction(liveaction_db):
+    """
+    Splits the liveaction into an ActionExecution compatible dict.
+    """
+    decomposed = {'liveaction': {}}
+    liveaction_api = vars(LiveActionAPI.from_model(liveaction_db))
+    for k in liveaction_api.keys():
+        if k in SKIPPED:
+            decomposed['liveaction'][k] = liveaction_api[k]
+        else:
+            decomposed[k] = getattr(liveaction_db, k)
```

Comment:
I hate this. Since LiveAction is a DictField in execution db, we need not have db representation of values. For other fields like start_timestamp, end_timestamp, we should use db representations. This code is error prone. I think we need to pause here and think this through. Otherwise, we are gonna have problems maintaining this.